File: educational-system-consumer-student-9001/src/main/java/edu/hubu/consumerstudent/service/IStudentService.java
package edu.hubu.consumerstudent.service;

import edu.hubu.commons.model.CommonResult;
import edu.hubu.commons.model.student.Student;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.*;

/**
 * @author moonlan
 * date 2021/2/1 下午4:54
 */
@SuppressWarnings(value = "unused")
@FeignClient(name = "student-provider-service")
public interface IStudentService {

    @GetMapping("/student/{id}")
    CommonResult selectOne(@PathVariable("id") String id);

    @GetMapping("/student/all")
    CommonResult selectAll(@RequestParam(value = "orderBy", defaultValue = "studentId") String orderBy);

    @GetMapping("/student/")
    CommonResult selectAllPage(
            @RequestParam(value = "currentPage", defaultValue = "1") Integer currentPage,
            @RequestParam(value = "pageSize", defaultValue = "5") Integer pageSize,
            @RequestParam(value = "orderBy", defaultValue = "studentId") String orderBy
    );

    @PostMapping("/student/select-like")
    CommonResult selectLikePage(
            @RequestParam(value = "currentPage", defaultValue = "1") Integer currentPage,
            @RequestParam(value = "pageSize", defaultValue = "5") Integer pageSize,
            @RequestParam(value = "orderBy", defaultValue = "studentId") String orderBy,
            @RequestBody edu.hubu.commons.mybatis.model.Student student
    );

    @PostMapping("/student/")
    CommonResult save(@RequestBody Student student);

    @PutMapping("/student/{id}")
    CommonResult update(@PathVariable("id") String id, @RequestBody Student student);

    @DeleteMapping("/student/{id}")
    CommonResult logicDeleteById(@PathVariable("id") String id);

    @PatchMapping("/student/{id}")
    CommonResult forbidById(@PathVariable("id") String id);

    @DeleteMapping("/student/delete/{id}")
    CommonResult deleteById(@PathVariable("id") String id);
}

File: educational-system-commons/src/main/java/edu/hubu/commons/service/student/IStudentService.kt
package edu.hubu.commons.service.student

import edu.hubu.commons.model.student.Student;

/**
* @author moonlan
* date 2021/1/25 下午2:07
*/
interface IStudentService {
    fun save(t: Student): Student
    fun logicalDeleteById(id: String): Boolean
    fun deleteById(id: String): Boolean
    fun updateById(id: String, t: Student): Student
    fun forbidById(id: String): Boolean
    fun selectOneById(id: String): Student
    fun selectAllOrderBy(order: String = "studentId"): List<Student>
    fun selectAllPageOrderBy(currentPage: Int = 1, pageSize: Int = 5, order: String = "studentId"): List<Student>
    fun selectLikePageOrderBy(currentPage: Int = 1, pageSize: Int = 5, order: String = "studentId", t: edu.hubu.commons.mybatis.model.Student): List<edu.hubu.commons.mybatis.model.Student>
}
